src/main/java/com/dansieproject/songr/controllers/HomeController.java

```diff
@@ -47,6 +45,42 @@ public RedirectView addTheAlbums(String albumName, String artistName, int songCo
         return new RedirectView("/album");
     }
 
+    @PostMapping("/album/delete")
+    public RedirectView deleteTheAlbum(long id){
+        albumRepository.deleteById(id);
+        return new RedirectView("/album");
+    }
+
+    @GetMapping("/album/{albumId}")
+    public String getAlbumDetails(Model m,@PathVariable long albumId){
+//        Album allie = albumRepository.findById(albumId);
+//        m.addAttribute("album", albumRepository.findById(albumId).get());
+        List<Song> songs = songRepository.findAll();
+        m.addAttribute("album", albumRepository.getOne(albumId));
+        m.addAttribute("songs", songs);
+        return "details";
+    }
+
+//    public Song(Album album, String title, int lengthSeconds, int trackNumber) {
+
+
+    @PostMapping("/album/{albumId}")
```

Comment:
You should most likely put the creation of a song on the Song route. It feels like the song might belong to an album but it is still a song.